jsk_panda_robot/panda_eus/euslisp/panda-interface.l

```diff
@@ -0,0 +1,237 @@
+     (send goal :goal :epsilon :outer outer) ;; [m]
+     ;;
+     (send act :send-goal goal)
+     (if wait (send act :wait-for-result))
+     ))
+  (:send-gripper-homing-action
+   (act &key (wait t))
+   (let ((goal (instance franka_gripper::HomingActionGoal :init)))
+     (send goal :header :stamp (ros::time-now))
+     ;;
+     (send act :send-goal goal)
+     (if wait (send act :wait-for-result))
+     ))
+  (:send-gripper-move-action
+   (act width speed &key (wait t))
+   (let ((goal (instance franka_gripper::MoveActionGoal :init)))
+     (send goal :header :stamp (ros::time-now))
+     (send goal :goal :width width) ;; [m]
+     (send goal :goal :speed speed) ;; [m/s]
+     ;;
+     (send act :send-goal goal)
+     (if wait (send act :wait-for-result))
+     ))
+  (:send-gripper-stop-action
+   (act &key (wait t))
+   (let ((goal (instance franka_gripper::StopActionGoal :init)))
+     (send goal :header :stamp (ros::time-now))
+     ;;
+     (send act :send-goal goal)
+     (if wait (send act :wait-for-result))
+     ))
+  (:stop-gripper
+   (arm &key (wait nil))
+   (let (acts)
+     (case
+         arm
+       (:arm (setq acts (list gripper-stop-action)))
+       )
+     (when acts
+       (dolist (act acts)
+         (send self :send-gripper-stop-action act
+               pos (/ (* 1000 0.08) tm) :wait nil))
+       (if wait (mapcar #'(lambda (act) (send act :wait-for-result)) acts))
+       )))
+  (:homing-gripper
+   (arm &key (wait nil))
+   (let (acts)
+     (case
+         arm
+       (:arm (setq acts (list gripper-homing-action)))
+       )
+     (when acts
+       (dolist (act acts)
+         (send self :send-gripper-homing-action act :wait nil))
+       (if wait (mapcar #'(lambda (act) (send act :wait-for-result)) acts))
+       )))
+  (:start-grasp
+   (arm &optional (pos 0.0) &key (effort 10.0) (tm 500) (wait nil)) ;; TODO :arms is not implemented
+   (let (acts)
+     (case
+         arm
+       (:arm (setq acts (list gripper-grasp-action)))
+       )
+     (when acts
+       (dolist (act acts)
+         (send self :send-gripper-grasp-action act
+               pos (/ (* 1000 0.08) tm) effort :wait nil))
+       (if wait (mapcar #'(lambda (act) (send act :wait-for-result)) acts))
+       )))
+  (:stop-grasp
+   (arm &key (wait nil)) ;; TODO :arms is not implemented
+   (unless (memq arm '(:arm :arms))
+     (error "you must specify arm ~A from ~A" (car args) '(:arm :arms))
+     (return-from :stop-grasp nil))
+   (send self :move-gripper arm 0.08 :tm 500 :wait wait)
+   )
+  (:move-gripper
+   (arm pos &key (tm 500) (timeout 5000) (wait nil))  ;; TODO :arms is not implemented
+   (let (acts)
+     (case
+         arm
+       (:arm (setq acts (list gripper-move-action)))
+       )
+     (when acts
+       (dolist (act acts)
+         (send self :send-gripper-move-action act
+               pos (/ (* 1000 0.08) tm) :wait nil))
+       (if wait (mapcar #'(lambda (act) (send act :wait-for-result)) acts))
+       )))
+  )
```

Comment:
Many parts are common to https://github.com/jsk-ros-pkg/jsk_robot/blob/93919243019ae06be43874022062abbeee00349e/jsk_panda_robot/panda_eus/euslisp/dual_panda-interface.l, so we have to re-use code in some way.
I'll make a PR for this after the other review comments are resolved or replied.

Reply:
That's great idea to extract common parts from the two interfaces.
I tried to think about it but it seemed difficult than I thought because a specific robot-model is required for the robot-interface of it here.
https://github.com/jsk-ros-pkg/jsk_robot/pull/1799/files/c943a29e24f7d8175393321b7f215e91d0b69829#diff-49407bb0669ccb3405b9c922a92547506aea3393e03e7c9aa2615efe645ed989R18

I hope to hear how this will be solved.

Reply:
@yuki-asano Currently I am working on this and want to ask you a question.
You set limb name as `:larm`:
https://github.com/jsk-ros-pkg/jsk_robot/pull/1799/files#diff-75f2fe8730d1b6c5ca7bf942abc99bf5955664a80692a4e7263428b14268a1c2R1
https://github.com/jsk-ros-pkg/jsk_robot/pull/1799/files#diff-75f2fe8730d1b6c5ca7bf942abc99bf5955664a80692a4e7263428b14268a1c2R10
But in euslisp common sense, single arm should be `:rarm`:
https://github.com/jsk-ros-pkg/jsk_robot/blob/93919243019ae06be43874022062abbeee00349e/jsk_denso_robot/cobottaeus/cobotta.yaml#L5
https://github.com/jsk-ros-pkg/jsk_robot/blob/93919243019ae06be43874022062abbeee00349e/jsk_fetch_robot/fetcheus/fetch.yaml#L7
https://github.com/jsk-ros-pkg/jsk_robot/blob/93919243019ae06be43874022062abbeee00349e/jsk_kinova_robot/kinovaeus/gen3_lite_gen3_lite_2f.yaml#L5

Can I take this opportunity to change the name to `:rarm`?
I'm afraid your code calls `:larm` many times.

Reply:
Thank you for telling the common sense of euslisp coding. I will check it out.

Reply:
change `:larm` to `:rarm`
[74fc6e4](https://github.com/jsk-ros-pkg/jsk_robot/pull/1799/commits/74fc6e4a871e547dc3855c0f20ab4c0f2ec66c6a)

Reply:
@yuki-asano Sorry for late, I completed
> extract common parts from the two interfaces

Could you check https://github.com/yuki-asano/jsk_robot/pull/1?

Reply:
Finally resolved by https://github.com/yuki-asano/jsk_robot/pull/2